sdk/resource/resource.go

```diff
@@ -29,7 +31,13 @@ type Resource struct {
 	labels label.Set
 }
 
-var emptyResource Resource
+var (
+	emptyResource Resource
+
+	defaultResource *Resource = func(r *Resource, err error) *Resource { return r }(
+		Detect(context.Background(), defaultServiceNameDetector{}, TelemetrySDK{}),
```

Comment:
Should we surface the returned error from `Detect` with the `ErrorHandler`?

Reply:
We can.

In the current implementation of `defaultServiceNameDetector` and `TelemetrySDK`, II don't think it's possible to return a non-nil error, but for future-proofing, it's probably good to call `otel.Handle(err)`